common-app-lib/src/cmr/common_app/services/search/parameters/converters/nested_field.clj

```diff
@@ -51,18 +56,59 @@
     (qm/string-condition (nested-field->elastic-keyword parent-field subfield) value
                          case-sensitive? pattern?)))
 
+(def subfield-name->condition-type
+  "Defines the subfield-name to non-string condition type mapping"
+  {:pass :int
+   ;; tiles could be comma separated string or list
+   :tiles :list})
+
+(defmulti nested-field+value->condition
+  "Converts a nested field and its value into a condition"
+  (fn [parent-field subfield-name subfield-value case-sensitive? pattern?]
+    (subfield-name->condition-type subfield-name)))
+
+(defmethod nested-field+value->condition :int
+  [parent-field subfield-name subfield-value _ _]
+  (qm/numeric-value-condition
+   (nested-field->elastic-keyword parent-field subfield-name)
+   subfield-value))
+
+(defn- comma-separated-string->list
+  "Returns the list of values from a comma separated string"
+  [value]
+  (map string/trim (string/split value #",")))
+
+(defn- normalized-list-value
+  "Returns the normalized value of a comma separated string or a list of such string."
+  [value]
+  (let [values (if (sequential? value)
+                 (mapcat comma-separated-string->list value)
+                 (comma-separated-string->list value))]
+    (distinct values)))
+
+(defmethod nested-field+value->condition :list
+  [parent-field subfield-name subfield-value case-sensitive? pattern?]
+  (let [normalized-values (normalized-list-value subfield-value)]
+    (nested-field+value->string-condition
+     parent-field subfield-name normalized-values case-sensitive? pattern?)))
+
+(defmethod nested-field+value->condition :default
+  [parent-field subfield-name subfield-value case-sensitive? pattern?]
+  (nested-field+value->string-condition
+   parent-field subfield-name subfield-value case-sensitive? pattern?))
+
 (defn parse-nested-condition
   "Converts a nested condition into a nested query model condition."
   [parent-field query-map case-sensitive? pattern?]
   (qm/nested-condition
-    parent-field
-    (gc/and-conds
-      (map (fn [[subfield-name subfield-value]]
-             (if (= :any subfield-name)
-               (gc/or-conds
-                 (map #(nested-field+value->string-condition parent-field % subfield-value
-                                                             case-sensitive? pattern?)
-                      (get-subfield-names parent-field)))
-               (nested-field+value->string-condition parent-field subfield-name subfield-value
-                                                     case-sensitive? pattern?)))
-           (dissoc query-map :pattern :ignore-case)))))
+   parent-field
```

Comment:
Nit: the original indentation was correct for our style - https://github.com/bbatsov/clojure-style-guide#one-space-indent.

Reply:
Can't read diffs correctly. You actually fixed it. 🙇

Reply:
Cool.